src/gc/impl/conservative/gc.d

```diff
@@ -2414,12 +2457,39 @@ struct Gcx
                 start = stop;
             }
 
-            markAll(nostack);
+            if (block)
+            {
+                markAll(nostack);
+            }
+            else
+            {
+                // fork now and sweep later
+                fflush(null); // avoid duplicated FILE* output
+                auto pid = fork();
+                assert(pid != -1); // TODO handle case
+                switch (pid)
+                {
+                    case -1: // fork() failed, retry without forking
+                        markProcPid = 0;
+                        return fullcollect(nostack, true);
```

Comment:
Same about the recursive call instead of just continuing with this call. Is this really safe? Maybe explain why in a comment?